This fixes the following compilation error:

    pdftexini.c: In function 'loadfmtfile':
    pdftexini.c:4016:25: error: assignment to 'boolean *' {aka 'int *'} from incompatible pointer type 'internalfontnumber *' {aka 'long int *'} [-Wincompatible-pointer-types]
     4016 |     pdffonthasspacechar = xmallocarray ( internalfontnumber , fontmax ) ;
          |                         ^
    pdftexini.c: In function 'mainbody':
    pdftexini.c:5569:27: error: assignment to 'boolean *' {aka 'int *'} from incompatible pointer type 'internalfontnumber *' {aka 'long int *'} [-Wincompatible-pointer-types]
     5569 |       pdffonthasspacechar = xmallocarray ( internalfontnumber , fontmax ) ;
          |                           ^
--- a/texk/web2c/pdftexdir/pdftex.ch	2024-10-04 22:50:00.127135615 +0200
+++ b/texk/web2c/pdftexdir/pdftex.ch	2024-10-04 23:08:11.623632667 +0200
@@ -179,7 +179,7 @@ pdf_font_type:=xmalloc_array(eight_bits,
 pdf_font_attr:=xmalloc_array(str_number, font_max);
 pdf_font_blink:=xmalloc_array(internal_font_number, font_max);
 pdf_font_elink:=xmalloc_array(internal_font_number, font_max);
-pdf_font_has_space_char:=xmalloc_array(internal_font_number, font_max);
+pdf_font_has_space_char:=xmalloc_array(boolean, font_max);
 pdf_font_stretch:=xmalloc_array(integer, font_max);
 pdf_font_shrink:=xmalloc_array(integer, font_max);
 pdf_font_step:=xmalloc_array(integer, font_max);
@@ -298,7 +298,7 @@ pdf_font_type:=xmalloc_array(eight_bits,
 pdf_font_attr:=xmalloc_array(str_number,font_max);
 pdf_font_blink:=xmalloc_array(internal_font_number,font_max);
 pdf_font_elink:=xmalloc_array(internal_font_number,font_max);
-pdf_font_has_space_char:=xmalloc_array(internal_font_number,font_max);
+pdf_font_has_space_char:=xmalloc_array(boolean,font_max);
 pdf_font_stretch:=xmalloc_array(integer,font_max);
 pdf_font_shrink:=xmalloc_array(integer,font_max);
 pdf_font_step:=xmalloc_array(integer,font_max);
